README.md

```diff
@@ -281,7 +287,211 @@ Examples of WordPress block markup and the associated data structure returned by
+
+#### Usage
+
+The `blocksData` field would be the field through which block data would be returned under a category that supports it. 
+
+An example of what a query would look like for a post:
+
+```graphQL
+query NewQuery {
+  post(id: "1", idType: DATABASE_ID) {
+    blocksData {
+      blocks {
+        id
+        name
+        attributes {
+          name
+          value
+        }
+        innerBlocks {
+          name
+          parentId
+          id
+          attributes {
+            name
+            value
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+Here, the `id` and `parentId` fields are dynamically generated, unique IDs that help to identify parent-child relationships in the `innerBlocks` under a block, in the overall block structure.The resulting `innerBlocks` is a flattened list, that can be untangled using the combination of `id` and `parentId` fields. This is helpful in being able to give back a complicated nesting structure, without having any knowledge as to how deep this nesting goes. 
+
+In addition, the attributes of a block are a list of `name`, `value` pairs so as to avoid having to figure out the type of each block. This allows giving back any block's attributes.
+
+#### Examples
+
+Examples of WordPress block markup and the associated data structure returned by the Block Data API.
+
+##### Example: Simple nested blocks: `core/list` and `core/quote`
+
+![List and Quote block in editor][media-example-list-quote]
+
+<table>
+<tr>
+<td>Block Markup</td>
+<td>Query</td>
+<td>Block Data API</td>
+</tr>
+<tr>
+<td>
+
+```html
+<!-- wp:list -->
+<ul><!-- wp:list-item -->
+  <li>This is item 1 in the list</li>
+  <!-- /wp:list-item -->
+
+  <!-- wp:list-item -->
+  <li>This is item 2 in the list</li>
+  <!-- /wp:list-item -->
+</ul>
+<!-- /wp:list -->
+
+<!-- wp:quote -->
+<blockquote class="wp-block-quote">
+  <!-- wp:paragraph -->
+  <p>This is a paragraph within a quote</p>
+  <!-- /wp:paragraph -->
+</blockquote>
+<!-- /wp:quote -->
+```
+
+</td>
+<td>
+
+```graphQL
+query NewQuery {
+  post(id: "1", idType: DATABASE_ID) {
+    blocksData {
+      blocks {
+        id
+        name
+        attributes {
+          name
+          value
+        }
+        innerBlocks {
+          name
+          parentId
+          id
+          attributes {
+            name
+            value
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+</td>
+<td>
+
+```json
+{
+  "data": {
+      "post": {
+          "blocksData": {
+              "blocks": [
+                  {
+                      "attributes": [
+                          {
+                              "name": "ordered",
+                              "value": ""
+                          },
+                          {
+                              "name": "values",
+                              "value": ""
+                          }
+                      ],
+                      "id": "1",
```

Comment:
Use example Relay IDs?

Reply:
I think I'm leaning toward keeping these as simplified IDs for the example. Actual IDs returned from GraphQL look super similar (`SUQ6MQ==`, `SUQ6Mg==`, `SUQ6MTY=`) and that gets in the way of demonstrating the connection between blocks and their parents. I tried subbing in some real values, but without easily visible relationships the code is a lot harder to follow:

```json
{
  "data": {
    "post": {
      "blocksData": {
        "blocks": [
          {
            "attributes": [
              {
                "name": "ordered",
                "value": ""
              },
              {
                "name": "values",
                "value": ""
              }
            ],
            "id": "SUQ6MjE=",
            "name": "core/list",
            "innerBlocks": [
              {
                "id": "SUQ6MTk=",
                "name": "core/list-item",
                "parentId": "SUQ6MjE=",
                "attributes": [
                  {
                    "name": "content",
                    "value": "This is item 1 in the list"
                  }
                ]
              },
              {
                "id": "SUQ6MTg=",
                "name": "core/list-item",
                "parentId": "SUQ6MjE=",
                "attributes": [
                  {
                    "name": "content",
                    "value": "This is item 2 in the list"
                  }
                ]
              }
            ]
          },
          {
            "attributes": [
              {
                "name": "value",
                "value": ""
              },
              {
                "name": "citation",
                "value": ""
              }
            ],
            "id": "SUQ6MTY=",
            "name": "core/quote",
            "innerBlocks": [
              {
                "id": "SUQ6MQ==",
                "name": "core/paragraph",
                "parentId": "SUQ6MTY=",
                "attributes": [
                  {
                    "name": "content",
                    "value": "This is a paragraph within a quote"
                  },
                  {
                    "name": "dropCap",
                    "value": ""
                  }
                ]
              }
            ]
          }
        ]
      }
    }
  }
}
```

Reply:
Got it, but the API will return Relay IDs, right? That's important since many client implementations used IDs as cache object identifiers. Using `int`s will result in collisions.

Reasoning makes sense, you could also put the decoded ID in a comment, like

```
"id": "SUQ6MQ==", // "BlockData:1"
```

But good either way

Reply:
That's correct that these are Relay IDs, and I agree that assuming integers could cause some issues. I tried out your ID commenting suggestion but felt like that also made the example too busy. Settled on a note underneath the example in https://github.com/Automattic/vip-block-data-api/pull/51/commits/90c77dedc6472b01e70226dfd4529b358a011fd4.